task6__israel_jacob/src/main.cpp

```diff
@@ -0,0 +1,149 @@
+#include"Cities.h"
+#include<iostream>
+#include<vector>
+#include <iostream>
+
+std::string getName(Cities cities);
+double getRadius();
+int getNorm();
+
+int main()
+{
+	Cities cities = Cities();
+	// Create an instance of the Cities class
+
+	try {
+		cities.addFromFile("data.txt");
+		// Attempt to load city data from the "data.txt" file
+	}
+	catch (const std::exception& e) {
+		std::cout << e.what();
+		return 0;
+	}
+
+	std::vector<std::pair<std::string, Location>> nearCities;
+	// Vector to store the nearby cities
+	int numOfCitiesInNorth;
+	// Variable to store the number of cities to the north
+	std::string name;
+	double radius;
+	int norm;
+
+	name = getName(cities);
+	// Get the name of the selected city from the user
+	while (name != "0")
+	{
+		try {
+		// Get the desired search radius from the user
+		radius = getRadius();
+
+		// Get the desired norm from the user
+		norm = getNorm();
+		}
+		catch (std::exception& e)
+		{
+			std::cout << e.what() << std::endl;
+			return 0;
+		}
+		nearCities = cities.nearCities(name, radius, norm);
```

Comment:
main is to complex, it handles reading inputs, searching and printing results

Reply:
Fixed it